pkg/cloudprovider/aws/aws.go

```diff
@@ -133,89 +142,165 @@ func (p *provider) TerminateInstance(providerID string) error {
 	return err
 }
 
-// ID returns the ID for the Autoscaling group
-func (a *autoscalingGroup) ID() string {
-	return aws.StringValue(a.group.AutoScalingGroupName)
+// Instances returns a map of all instances in the Autoscaling group
+// with providerID as key and cloudprovider.Instance as value
+func (a *autoscalingGroups) Instances() map[string]cloudprovider.Instance {
+	instances := make(map[string]cloudprovider.Instance)
+	for _, group := range a.groups {
+		for _, i := range group.Instances {
+			providerID, err := instanceIDToProviderID(*i.InstanceId, *i.AvailabilityZone)
+			if err != nil {
+				a.logger.Info("[Instances] skip instance which failed instanceID to providerID conversion: %v", *i.InstanceId)
+				continue
+			}
+			instances[providerID] = &instance{
+				instance:      i,
```

Comment:
Will I get caught by the "for iterator" gotcha in Go? E.g. `i` always will end up pointing to the last element since it's not copied. Seems to be okay based on diff but just checking

Reply:
No. That problem was due to iterate a list of objects, copy the object into a temporary variable and append that temporary variable address into a list which is always the same. This one is a list of object pointer and the temporary variable `i` is also a pointer which will be updated for each iteration and that pointer will be added to that instance object.